src/generators/gradient_based/base.jl

```diff
@@ -29,12 +29,12 @@ end
 
 Default outer constructor for `Generator`.
 """
-function Generator(;
+function GradientBasedGenerator(;
```

Comment:
It would be nice if the comment explained the fields of the struct I think?

Reply:
Sure, I'll add that.